packages/_flutterfire_internals/lib/_flutterfire_internals.dart

```diff
@@ -56,11 +58,23 @@ FirebaseException _firebaseExceptionFromCoreFirebaseError(
   required String Function(String) codeParser,
   required String Function(String code, String message)? messageParser,
 }) {
-  // ignore: unnecessary_cast
-  final convertCode = firebaseError.code as String;
+  // Necessary because of the conditional import
+  late final String convertCode;
+  if ((firebaseError.code) is JSAny) {
```

Comment:
```suggestion
  if (firebaseError.code is JSAny) {
```
I think the extra parentheses can be removed